src/main/java/org/example/gabartischolarsqbbe/service/JobService.java

```diff
@@ -0,0 +1,100 @@
+package org.example.gabartischolarsqbbe.service;
+
+import org.example.gabartischolarsqbbe.dto.JobRequest;
+import org.example.gabartischolarsqbbe.dto.JobResponse;
+import org.example.gabartischolarsqbbe.entity.Job;
+import org.example.gabartischolarsqbbe.repository.JobRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
+public class JobService {
+    
+    private final JobRepository jobRepository;
+    
+    @Autowired
+    public JobService(JobRepository jobRepository) {
+        this.jobRepository = jobRepository;
+    }
+    
+    /**
+     * Create a new job
+     * @param jobRequest the job request containing name and code
+     * @return JobResponse with the created job details
+     * @throws IllegalArgumentException if job code already exists
+     */
+    public JobResponse createJob(JobRequest jobRequest) {
```

Comment:
I think it would be better to have a `DTO` for JobRequest` input, its use might not be too obvious now but it would be important in case the code and the entity grows in fields and logic